### ISSUE TITLE
Squiz.Commenting.FunctionComment does not work with generics

### ISSUE THREAD

**Issue by simPod** [squizlabs/PHP_CodeSniffer#2513]
```php
    /**
     * @param string $namespace e.g. cms
     * @param string $name      e.g. duration_seconds
     * @param string $help      e.g. The duration something took in seconds.
     * @param iterable<string> $labels e.g. ['controller', 'action']
     */
```

Last line says
> Expected 4 spaces after parameter name; 1 found (Squiz.Commenting.FunctionComment.SpacingAfterParamName)

Expected working example is this
```php
    /**
     * @param string $namespace        e.g. cms
     * @param string $name             e.g. duration_seconds
     * @param string $help             e.g. The duration something took in seconds.
     * @param iterable<string> $labels e.g. ['controller', 'action']
     */
```

But that complains:

>  Expected 1 spaces after parameter name; 8 found (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
> Expected 6 spaces after parameter name; 13 found (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
> Expected 6 spaces after parameter name; 13 found (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
> Expected 4 spaces after parameter name; 1 found (Squiz.Commenting.FunctionComment.SpacingAfterParamName)

IMO it ignores `iterable<>` token and counts `string` token only

**Comment by simPod** [comment]
Also this is fails
```php
    /**
     * @param BigNumber|number|string $amount The monetary amount.
     * @param Currency|string|int $currency   The Currency instance, ISO currency code or ISO numeric currency code.
     */
```

**Comment by rodrigoprimo** [comment]
I believe this report is no longer valid (or never was) and can be closed. Unless I'm missing something, the sniff correctly handles alignment/spacing in the examples provided.

The sniff correctly treats `iterable<string>` (16 chars) as longer than `string` (6 chars), and expects the `string` params to be padded with extra spaces to align with it.

For reference, here is how PHPCS currently handles the code samples from the issue and its comment (I adapted the samples slightly to avoid unrelated errors).

**Sample 1** (from the issue description):
```php
/**
 * @param string $namespace E.g. cms.
 * @param string $name      E.g. duration_seconds.
 * @param string $help      E.g. the duration something took in seconds.
 * @param iterable<string> $labels E.g. ['controller', 'action'].
 *
 * @return void
 */
function test(string $namespace, string $name, string $help, iterable $labels) {}
```

Running `phpcs -s` correctly flags the alignment issues:
```
  4 | ERROR | [x] Expected 11 spaces after parameter type; 1 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamType)
  5 | ERROR | [x] Expected 11 spaces after parameter type; 1 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamType)
  6 | ERROR | [x] Expected 11 spaces after parameter type; 1 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamType)
  7 | ERROR | [x] Expected 4 spaces after parameter name; 1 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
 11 | ERROR | [ ] Expected type hint "iterable<string>"; found "iterable" for
    |       |     $labels (Squiz.Commenting.FunctionComment.IncorrectTypeHint)
```

**Sample 2** (the "expected working example" from the issue):
```php
/**
 * @param string $namespace        E.g. cms.
 * @param string $name             E.g. duration_seconds.
 * @param string $help             E.g. the duration something took in seconds.
 * @param iterable<string> $labels E.g. ['controller', 'action'].
 *
 * @return void
 */
function test(string $namespace, string $name, string $help, iterable $labels) {
}
```

Running `phpcs -s` flags both the type and name alignment issues:
```
  4 | ERROR | [x] Expected 11 spaces after parameter type; 1 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamType)
  4 | ERROR | [x] Expected 1 spaces after parameter name; 8 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
  5 | ERROR | [x] Expected 11 spaces after parameter type; 1 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamType)
  5 | ERROR | [x] Expected 6 spaces after parameter name; 13 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
  6 | ERROR | [x] Expected 11 spaces after parameter type; 1 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamType)
  6 | ERROR | [x] Expected 6 spaces after parameter name; 13 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
  7 | ERROR | [x] Expected 4 spaces after parameter name; 1 found
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
 11 | ERROR | [ ] Expected type hint "iterable<string>"; found "iterable" for
    |       |     $labels (Squiz.Commenting.FunctionComment.IncorrectTypeHint)
```

PHPCBF auto-fixes both samples to the same correctly aligned result:
```php
/**
 * @param string           $namespace E.g. cms.
 * @param string           $name      E.g. duration_seconds.
 * @param string           $help      E.g. the duration something took in seconds.
 * @param iterable<string> $labels    E.g. ['controller', 'action'].
 *
 * @return void
 */
function test(string $namespace, string $name, string $help, iterable $labels) {}
```

**Sample 3** (from the [comment](https://github.com/squizlabs/PHP_CodeSniffer/issues/2513#issuecomment-495181498) on this issue):
```php
/**
 * @param BigNumber|number|string $amount The monetary amount.
 * @param Currency|string|int $currency   The Currency instance, ISO currency code or ISO numeric currency code.
 *
 * @return void
 */
function test(string $amount, string $currency) {}
```

Running `phpcs -s`:
```
 4 | ERROR | [x] Expected 3 spaces after parameter name; 1 found
   |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
 5 | ERROR | [x] Expected "Currency|string|integer" but found
   |       |     "Currency|string|int" for parameter type
   |       |     (Squiz.Commenting.FunctionComment.IncorrectParamVarName)
 5 | ERROR | [x] Expected 5 spaces after parameter type; 1 found
   |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamType)
 5 | ERROR | [x] Expected 1 spaces after parameter name; 3 found
   |       |     (Squiz.Commenting.FunctionComment.SpacingAfterParamName)
```

PHPCBF auto-fixes to:
```php
/**
 * @param BigNumber|number|string $amount   The monetary amount.
 * @param Currency|string|integer $currency The Currency instance, ISO currency code or ISO numeric currency code.
 *
 * @return void
 */
function test(string $amount, string $currency): void {}
```

Tested on the `4.x` branch.

Note: both samples 1 and 2 also trigger an `IncorrectTypeHint` error (`Expected type hint "iterable<string>"; found "iterable" for $labels`). That is a separate issue tracked in PHPCSStandards/PHP_CodeSniffer#746.
